src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * Creates a command that sets the target position of the arm.
+     *
+     * @param targetAngle             the target angle of the arm
+     * @param targetElevatorPosition  the target position of the arm's elevator
+     * @param angleSpeedPercentage    the percentage of speed that the angle will move in
+     * @param elevatorSpeedPercentage the percentage of speed that the elevator will move in
+     * @return the command
+     */
+    public Command getSetTargetArmPositionCommand(Rotation2d targetAngle, double targetElevatorPosition, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        return new DeferredCommand(
+                () -> getCurrentSetTargetArmStateCommand(targetAngle, targetElevatorPosition, angleSpeedPercentage, elevatorSpeedPercentage),
+                Set.of(this)
+        );
+    }
+
+    private Command getCurrentSetTargetArmStateCommand(Rotation2d angle, double elevatorPosition, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        if (isElevatorOpening(elevatorPosition)) {
+            return new SequentialCommandGroup(
+                    getSetTargetAngleCommand(angle, angleSpeedPercentage),
+                    getSetTargetElevatorPositionCommand(elevatorPosition, elevatorSpeedPercentage)
+            );
+        }
+        return new SequentialCommandGroup(
+                getSetTargetElevatorPositionCommand(elevatorPosition, elevatorSpeedPercentage),
+                getSetTargetAngleCommand(angle, angleSpeedPercentage)
+        );
+    }
+
+    private Command getSetTargetAngleCommand(Rotation2d targetAngle, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(targetAngle, speedPercentage),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private Command getSetTargetElevatorPositionCommand(double targetElevatorPosition, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(targetElevatorPosition, speedPercentage),
+                this::setTargetElevatorPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private boolean isElevatorOpening(double targetElevatorPosition) {
+        return getElevatorPositionRevolutions() < targetElevatorPosition;
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            armIO.stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        double voltage = calculateAngleMotorOutput(targetState);
+        armIO.setAnglePower(voltage);
```

Comment:
Have a function called of the io `setTargetAngle` that accepts a rotation2d and the ff and move the pid calculation to the io.

Reply:
I don't understand. u said that the logic is meant to be in the normal Arm class. and also, the IO doesn't have an arm inputs. do I need to make one?

Reply:
wdym by "a normal arm class" 
1. setting target position sometimes would be using onboard pid, so if we'll have another setup that uses onboard pid we'll need to set target position. It's not really logic and more motor setting. 
2. Yes, every IO class needs to have an inputs class in order to get info from the motors, though I don't see why this is relevant to this

Reply:
1. I mean the Arm class.
2. wdym? Do I need to have the method accept a pid controller so that I can use it with an onboard PID? adn wdym by motor setting? you said I should move everything in this method to the IO. but I can't put it in the base ArmIO class because it returns something, and I can't add it to the inputs class because it needs a state. So where do I put it?
3. nvm, I figured out my problem with the inputs

Reply:
The arm subsystem is supposed to control logic. So things like commands, motion profiles, etc. When it comes to motor setting (things like setting target power or position), these kinds of things need to be handled in the io implementation. 
So you'll have a function that accepts target position and velocity and runs the pid + ff calculations to set the target position. The pid controller and feedforward will be a part of the implementation and will be saved in the implementation's constants

Reply:
so I leave the part that calculates the pid and ff in the Arm class, but use the IO to power the motors acording to the calculation? that doesn't make sense as I don't have to make a separate method, I can just use the setAnglePower method like I already am. I don't get what u want me to add

Reply:
u also said to move the pid calculation into the io but doesn't that count as logic?